### PR TITLE
i18n(hi): add `components.mdx`

### PR DESCRIPTION
**Pull request by at-the-vr**
#### Description
- add translation for [Components](https://starlight.astro.build/hi/guides/components/) in hindi language

### CODE REVIEW

File: docs/src/content/docs/hi/guides/components.mdx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+</AnotherComponent>
+```
+
+क्योंकि Starlight Astro द्वारा संचालित है, आप अपने MDX फ़ाइलों में किसी भी [समर्थित UI फ्रेमवर्क (React, Preact, Svelte, Vue, Solid, Lit और Alpine)](https://docs.astro.build/hi/core-concepts/framework-components/) के साथ निर्मित अवयवों के लिए समर्थन जोड़ सकते हैं।
+Astro दस्तावेज़ीकरण में [MDX में अवयवों का उपयोग करने](https://docs.astro.build/hi/guides/markdown-content/#using-components-in-mdx) के बारे में और जानें।
+
+### Starlight की शैलियों के साथ संगतता
+
+Starlight आपके Markdown कंटेंट पर डिफ़ॉल्ट शैलि लागू करता है, उदाहरण के लिए तत्वों के बीच margin जोड़ना।
+यदि ये शैलियाँ आपके अवयव की उपस्थिति के साथ संघर्ष करती हैं, तो उन्हें अक्षम करने के लिए अपने अवयव पर 'not-content' वर्ग सेट करें।
```
Comment (HiDeoo) [inline docs/src/content/docs/hi/guides/components.mdx:39]:
```suggestion
यदि ये शैलियाँ आपके अवयव की उपस्थिति के साथ संघर्ष करती हैं, तो उन्हें अक्षम करने के लिए अपने अवयव पर `not-content` वर्ग सेट करें।
```

File: docs/src/content/docs/hi/guides/components.mdx [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+प्रत्येक `<TabItem>` में उपयोगकर्ताओं को प्रदर्शित करने के लिए एक `label` होना चाहिए।
+
+```mdx
+# src/content/docs/example.mdx
+
+import { Tabs, TabItem } from '@astrojs/starlight/components';
+
+<Tabs>
+	<TabItem label="Stars">Sirius, Vega, Betelgeuse</TabItem> //TODO
+	<TabItem label="Moons">Io, Europa, Ganymede</TabItem>
```
Comment (HiDeoo) [inline docs/src/content/docs/hi/guides/components.mdx:70]: I think the `//TODO` comment has been forgotten.

File: docs/src/content/docs/hi/guides/components.mdx [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+</CardGrid>
+```
+
+:::
+
+### लिंक कार्ड
+
+विभिन्न पेजों को प्रमुखता से लिंक करने के लिए `<LinkCard>` अवयव का उपयोग करें।
+
+`<LinkCard>` के लिए एक शीर्षक और एक [`href`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#href) विशेषता की आवश्यकता होती है। आप वैकल्पिक रूप से एक संक्षिप्त `description` या `target` जैसी अन्य लिंक विशेषताएँ शामिल कर सकते हैं।
```
Comment (HiDeoo) [inline docs/src/content/docs/hi/guides/components.mdx:136]:
```suggestion
`<LinkCard>` के लिए एक `title` और एक [`href`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#href) विशेषता की आवश्यकता होती है। आप वैकल्पिक रूप से एक संक्षिप्त `description` या `target` जैसी अन्य लिंक विशेषताएँ शामिल कर सकते हैं।
```

The `title` reference the attribute and should not be translated (not sure if my suggestion is correct tho).

**Review by HiDeoo — COMMENTED** [review]
Thanks for the translation :raised_hands: I also left a few small comments (I don't think they're duplicate with the ones from Chris).

File: docs/src/content/docs/hi/guides/components.mdx [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+import { Icon } from '@astrojs/starlight/components';
+
+<Icon name="star" color="goldenrod" size="2rem" />
+```
+
+उपरोक्त कोड पेज पर निम्नलिखित उत्पन्न करता है:
+
+<Icon name="star" color="goldenrod" size="2rem" />
+
+#### All icons
```
[the remark this answers is not in the document]
Comment (at-the-vr) [reply docs/src/content/docs/hi/guides/components.mdx:194]: Thanks for the feedback, all of them were useful (especially the headings one) 😄

File: docs/src/content/docs/hi/guides/components.mdx (hunk 3 above)
Comment (at-the-vr) [reply docs/src/content/docs/hi/guides/components.mdx:136]: Nice catch(s) 😄 Every single one is taken in account 👍

**Comment by HiDeoo** [comment]
Just a small note as I may not have been clear with the [comment](https://github.com/withastro/starlight/pull/1371/files/79a2e25c0ddfecf62f6061bdd58ddc8f382b4f98#r1452165317) regarding `//TODO` and I see that you just translated it. I think this is something you added and forgot to remove as this is not present in the [original content](https://github.com/withastro/starlight/blob/main/docs/src/content/docs/guides/components.mdx#L69).

**Comment by at-the-vr** [comment]
> Just a small note as I may not have been clear with the [comment](https://github.com/withastro/starlight/pull/1371/files/79a2e25c0ddfecf62f6061bdd58ddc8f382b4f98#r1452165317) regarding `//TODO` and I see that you just translated it. I think this is something you added and forgot to remove as this is not present in the [original content](https://github.com/withastro/starlight/blob/main/docs/src/content/docs/guides/components.mdx?rgh-link-date=2024-01-15T14%3A13%3A11Z#L69).

I am laughing and contemplating about what just happened, I am not using TODO terminologies for pending translation again

File: docs/src/content/docs/hi/guides/components.mdx [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+	</Card>
+	<Card title="चन्द्रमा" icon="moon">
+		Io, Europa, Ganymede
+	</Card>
+</CardGrid>
+```
+
+उपरोक्त कोड पेज पर निम्नलिखित उत्पन्न करता है:
+
+<Card title="इसको देखेF">दिलचस्प कंटेंट जिसे आप हाइलाइट करना चाहते हैं।</Card>
```
Comment (delucis) [inline docs/src/content/docs/hi/guides/components.mdx:109]: This is a typo right?

```suggestion
<Card title="इसको देखे">दिलचस्प कंटेंट जिसे आप हाइलाइट करना चाहते हैं।</Card>
```

**Review by delucis — COMMENTED** [review]
Thanks for the updates @at-the-vr — and he detailed review @HiDeoo, I totally missed those!

Just spotted one last thing but otherwise LGTM 🚀

**Review by delucis — APPROVED** [review]
Thanks again @at-the-vr — LGTM! 🚀

**Pull request merged**